src/Control/FunFlow/Base.hs

```diff
@@ -47,19 +46,19 @@ step = effect . Step
 named :: Store b => T.Text -> (a -> b) -> Flow eff ex a b
 named n f = effect $ Named n f
 
-external :: ContentHashable a => (a -> ExternalTask) -> Flow eff ex a ContentHash
+external :: ContentHashable a => (a -> ExternalTask) -> Flow eff ex a CS.Item
 external = effect . External
 
 wrap :: eff a b -> Flow eff ex a b
 wrap = effect . Wrapped
 
-docker :: ContentHashable a => (a -> Docker.Config) -> Flow eff ex a ContentHash
+docker :: ContentHashable a => (a -> Docker.Config) -> Flow eff ex a CS.Item
 docker f = external $ Docker.toExternal . f
 
-putInStore :: (ContentHashable a, Store a) => Flow eff ex a ContentHash
-putInStore = effect $ PutInStore
-getFromStore :: (ContentHashable a, Store a) => Flow eff ex ContentHash (Maybe a)
-getFromStore = effect $ GetFromStore
+putInStore :: ContentHashable a => (FilePath -> a -> IO ()) -> Flow eff ex a CS.Item
```

Comment:
I wonder at the generality of these signatures. I guess the advantage of them is that it's easy to read/write files in any way you desire. But on the other hand, these signatures can hide a lot of sins, including lazy I/O, which could give some confusing results (as well as maybe violating store semantics?).

Since, at least for the moment, we know we're going to be reading the whole contents to a strict `ByteString`, is there any reason not to have `a -> ByteString` and `ByteString -> Maybe a` here?

Reply:
My thinking was that when dealing with external tasks it might be necessary to control the filename and maybe provide, or read multiple files in a subtree. I agree that those signatures allow too much. Maybe we could use `Map FilePath ByteString` for a start?

Reply:
That's a good point. And of course we might want to allow traversing the directory tree (user might not know what files are there in advance).

I almost want to suggest writing a small monadic DSL for serialising/deserialising directory trees, but I think @glutamate would be upset with me :-)

Reply:
Haha, yes, I was thinking of that, too. It's very tempting. Though, it might be too much of a tangent at this point.